driver-core/src/main/com/mongodb/connection/Server.java

```diff
@@ -54,5 +54,5 @@
      *
      * @param callback the callback to execute when the connection is available or an error occurs
      */
-    void getConnectionAsync(SingleResultCallback<Connection> callback);
+    void getConnectionAsync(SingleResultCallback<AsyncConnection> callback);
```

Comment:
I would not go that far.   The only reason AsyncSingleConnectionBinding does that is because it's a test class.